components/inputs/input-fieldset.js

```diff
@@ -35,36 +41,41 @@ class InputFieldset extends PropertyRequiredMixin(InputInlineHelpMixin(SkeletonM
 	}
 
 	static get styles() {
-		return [ super.styles, inputLabelStyles, offscreenStyles,
+		return [ super.styles, heading4Styles, inputLabelStyles, offscreenStyles,
 			css`
 				:host {
 					display: block;
 				}
 				:host([hidden]) {
 					display: none;
 				}
+				legend.d2l-heading-4 {
+					margin-block: 0 0.9rem;
+					padding: 0;
```

Comment:
`<legend>`s come with left/right padding for some reason, so we have to negate it.